onnxruntime/core/providers/dnnl/dnnl_execution_provider.cc

```diff
@@ -398,22 +406,22 @@ void DNNLExecutionProvider::CreateMetaDef(const onnxruntime::GraphViewer& graph_
     }
   }
 
-  ONNX_NAMESPACE::AttributeProto ap;
-  ap.set_s(subgraph_id);
-  ap.set_type(ONNX_NAMESPACE::AttributeProto_AttributeType::AttributeProto_AttributeType_STRING);
-  meta_def->attributes["subgraph_id"] = ap;
-  std::unique_ptr<IndexedSubGraph> sub_graph = onnxruntime::make_unique<IndexedSubGraph>();
-  sub_graph->nodes = sub_var.subgraph_node_indexes;
+  auto ap = ONNX_NAMESPACE::Prov_AttributeProto::Create();
+  ap->set_s(subgraph_id);
+  ap->set_type(ONNX_NAMESPACE::AttributeProto_AttributeType::AttributeProto_AttributeType_STRING);
+  meta_def->attributes["subgraph_id"] = std::move(ap);
+  auto sub_graph = onnxruntime::Prov_IndexedSubGraph::Create();
+  sub_graph->Nodes() = sub_var.subgraph_node_indexes;
   sub_graph->SetMetaDef(meta_def);
-  result.push_back(onnxruntime::make_unique<ComputeCapability>(std::move(sub_graph)));
+  result.push_back(onnxruntime::make_unique<Prov_ComputeCapability>(std::move(sub_graph)));
   mkl_subgraphs_.insert(std::make_pair(subgraph_id, subgraph_ptr));
 
   // Reset subgraph and meta_Def
   sub_var.Reset();
 }
 
-Status DNNLExecutionProvider::Compile(const std::vector<onnxruntime::Node*>& fused_nodes,
-                                        std::vector<NodeComputeInfo>& node_compute_funcs) {
+Status DNNLExecutionProvider::Prov_Compile(const std::vector<onnxruntime::Prov_Node*>& fused_nodes,
```

Comment:
Any reason to use Prov_Compile and not just Compile?

Reply:
Unfortunately, yes. There's a class that inherits from both the IExecutionProvider and Prov_IExecutionProvider_Router to manage the inheritance so you can't override the same named function and have two separate implementations (even if the parameters don't match).

It's another thing I want to change further in the future as it's not pretty.